### PR TITLE
major/feature/UBI-542-addAuthMiddleware

### PR DESCRIPTION
**Pull request by JeremiSavard**
#### What changes does this PR contain?
add middleware to validate token
#### Which JIRA ticket ID's are associated with this PR?
ubi-542

### CODE REVIEW

File: chart/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: microservice-template
 type: application
 version: 0.0.0
-appVersion: 0.0.0
+appVersion: 0.0.0-rc
```
Comment (ysma500) [inline chart/Chart.yaml:5]: This shouldn't be a commited change

File: chart/values.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ replicaCount: 1
 
 image:
   repository: ubiviushub/microservice-template
-  tag: 0.0.0
+  tag: 0.0.0-rc
```
Comment (ysma500) [inline chart/values.yaml:5]: Same here

**Review by ysma500 — CHANGES_REQUESTED** [review]

File: go.mod [hunk 3]
```diff
@@ -4,16 +4,13 @@ go 1.15
 
 require (
 	github.com/Ubivius/pkg-telemetry v1.0.0
-	github.com/dvyukov/go-fuzz v0.0.0-20210914135545-4980593459a1 // indirect
-	github.com/elazarl/go-bindata-assetfs v1.0.1 // indirect
+	github.com/Ubivius/shared-authentication v0.0.0-20211113221950-120d47cecb60 // indirect
```
Comment (ArcticWalrus) [inline go.mod:7]: Should do a git tag for v1.0.0 or something in repo and import the exact version

File: pkg/router/router.go [hunk 4]
```diff
@@ -18,25 +19,30 @@ func New(productHandler *handlers.ProductsHandler) *mux.Router {
...
 	// Get Router
 	getRouter := router.Methods(http.MethodGet).Subrouter()
+	getRouter.Use(tokenValidation.Middleware)
 	getRouter.HandleFunc("/products", productHandler.GetProducts)
 	getRouter.HandleFunc("/products/{id:[0-9a-z-]+}", productHandler.GetProductByID)
 
 	//Health Check
-	getRouter.HandleFunc("/health/live", productHandler.LivenessCheck)
-	getRouter.HandleFunc("/health/ready", productHandler.ReadinessCheck)
+	healthRouter := router.Methods(http.MethodGet).Subrouter()
```
Comment (ArcticWalrus) [inline pkg/router/router.go:27]: Do you know if setting another http.Method get router stop the other get endpoints from being exposed

**Review by ArcticWalrus — CHANGES_REQUESTED** [review]

**Review by ysma500 — APPROVED** [review]
LGTM

**Review by ArcticWalrus — APPROVED** [review]

**Pull request merged**